pkg/apis/core/validation/validation.go

```diff
@@ -5441,7 +5441,7 @@ func ValidateServiceUpdate(service, oldService *core.Service) field.ErrorList {
 // ValidateServiceStatusUpdate tests if required fields in the Service are set when updating status.
 func ValidateServiceStatusUpdate(service, oldService *core.Service) field.ErrorList {
 	allErrs := ValidateObjectMetaUpdate(&service.ObjectMeta, &oldService.ObjectMeta, field.NewPath("metadata"))
-	allErrs = append(allErrs, ValidateLoadBalancerStatus(&service.Status.LoadBalancer, field.NewPath("status", "loadBalancer"))...)
+	allErrs = append(allErrs, ValidateLoadBalancerStatus(&service.Status.LoadBalancer, field.NewPath("status", "loadBalancer"), &service.Spec)...)
```

Comment:
why not check directly here?

```suggestion
if !utilfeature.DefaultFeatureGate.Enabled(features.AllowServiceLBStatusOnNonLB) && spec.Type != core.ServiceTypeLoadBalancer && len(status.Ingress) != 0 {
		allErrs = append(allErrs, field.Forbidden(ingrPath, "may only be used when `spec.type` is 'LoadBalancer'"))
		} else {
	allErrs = append(allErrs, ValidateLoadBalancerStatus(&service.Status.LoadBalancer, field.NewPath("status", "loadBalancer"), &service.Spec)...)
	}
```

Reply:
it felt more contained in the status-specific path.  Other than avoiding one argument, is there a reason I am not seeing?

Reply:
I was trying to avoid to plumb the spec on the status path

Reply:
my point is that if we make `service.Spec` an argument of `ValidateLoadBalancerStatus` we may open a door to all kind of validations there , in this case have `ValidateLoadBalancerStatus` that is can be called from any `serviceType`, I think that this is our problem, we allow it from any service type.

with my suggesting once we remove the feature gate the code seems easy to read, as `ValidateLoadBalancerStatus` only will validate the status for types LoadBalancer that is what you expect from the name